doc/api/esm.md

```diff
@@ -1892,8 +1892,10 @@ The resolver can throw the following errors:
 **READ_PACKAGE_SCOPE**(_url_)
 
 > 1. Let _scopeURL_ be _url_.
-> 1. While _scopeURL_ is not the file system root,
+> 1. Do-while _scopeURL_ is not the file system root,
```

Comment:
Perhaps let's be explicit and either make it `1. "Do,"`, with the `Loop while` part at the end of the loop block.

Reply:
That doesn't seem like a possibility seeing as how each line is numbered. You're probably thinking about http://krypton.mnsu.edu/~sj9525me/Academic/IS&T110/Lab/Psuedocode%20for%20control%20structures%20.pdf, but that style of pseudocode doesn't mesh well with this one unless I am misunderstanding.

Reply:
> That doesn't seem like a possibility seeing as how each line is numbered.

I'm not sure I follow. `1. Do,` as its own line before the block seems fine to me, and lines up with the expectations. ECMA-262 has many cases of `Repeat,` as its own line, although I couldn't find any `do-while` examples there.

Reply:
The alternative is to flatten the condition into the body of the loop of course.

Reply:
Are you concerned that it saying `Do-while` (like currently) isn't the proper way of expressing this in pseudocode? I don't have high confidence about this either (and my Google-fu is failing me here as well). I would be inclined to leave it as-is for the sake of clarity, but am unsure about the correctness.

/cc @ljharb

Reply:
Yes I am suggesting two tangible alternatives to make it clearer.

Reply:
Would you mind putting those as GitHub suggestion comments? I'm having trouble reifying your intention.

Reply:
Actually I think the following will work:

```
> 1. Let _scopeURL_ be _url_.
> 1. While _scopeURL_ is not the file system root,
>    1. Set _scopeURL_ to the parent URL of _scopeURL_.
>    1. If _scopeURL_ ends in a _"node_modules"_ path segment, return **null**.
>    1. Let _pjson_ be the result of **READ_PACKAGE_JSON**(_scopeURL_).
>    1. If _pjson_ is not **null**, then
>       1. Return _pjson_.
> 1. Return **null**.
```

because we shouldn't actually be checking `path/to/file.js/package.json` so I think that gets the ordering right in the spec here.